mpmath/ctx_mp.py

```diff
@@ -411,7 +411,7 @@ def workdps(ctx, n, normalize_output=False):
         """
         return PrecisionManager(ctx, None, lambda d: n, normalize_output)
 
-    def autoprec(ctx, f, maxprec=None, catch=(), verbose=False):
+    def autoprec(ctx, f, maxprec=None, catch=(), verbose=False, lr=2, early_termination=False):
```

Comment:
1. ``lr`` parameter name should be more verbose.  Or maybe we should instead offer a more generic API, e.g. a lambda-function that will be used to adjust precision?
2. I think that ``early_termination`` and ``det`` changes are out of the scope of this pr.